direct/src/gui/DirectFrame.py

```diff
@@ -50,166 +52,113 @@ def __init__(self, parent = None, **kw):
 
     def setText(self, text=None):
-        if text is not None:
-            self['text'] = text
-
-        # Determine if user passed in single string or a sequence
-        if self['text'] == None:
-            textList = (None,) * self['numStates']
-        elif isinstance(self['text'], stringType):
-            # If just passing in a single string, make a tuple out of it
-            textList = (self['text'],) * self['numStates']
-        else:
-            # Otherwise, hope that the user has passed in a tuple/list
-            textList = self['text']
-        # Create/destroy components
-        for i in range(self['numStates']):
-            component = 'text' + repr(i)
-            # If fewer items specified than numStates,
-            # just repeat last item
-            try:
-                text = textList[i]
-            except IndexError:
-                text = textList[-1]
-
-            if self.hascomponent(component):
-                if text == None:
-                    # Destroy component
-                    self.destroycomponent(component)
-                else:
-                    self[component + '_text'] = text
-            else:
-                if text == None:
-                    return
-                else:
-                    from .OnscreenText import OnscreenText
-                    self.createcomponent(
-                        component, (), 'text',
-                        OnscreenText,
-                        (), parent = self.stateNodePath[i],
-                        text = text, scale = 1, mayChange = self['textMayChange'],
-                        sort = DGG.TEXT_SORT_INDEX,
-                        )
+        self.new_set("text", text)
 
     def clearGeom(self):
         self['geom'] = None
         self.setGeom()
 
-    def setGeom(self, geom=None):
-        if geom is not None:
-            self['geom'] = geom
-
-        # Determine argument type
-        geom = self['geom']
-
-        if geom == None:
-            # Passed in None
-            geomList = (None,) * self['numStates']
-        elif isinstance(geom, NodePath) or \
-             isinstance(geom, stringType):
-            # Passed in a single node path, make a tuple out of it
-            geomList = (geom,) * self['numStates']
+    def long_type_check(self, name, object_):
+        if isinstance(object_, stringType):
+            return True
+        if name == "image":
+            if ((isinstance(object_, NodePath) or
+                isinstance(object_, Texture)) 
+                or
+                ((len(object_) == 2) and# Its a model/node pair of strings
+                isinstance(object_[0], stringType) and
+                isinstance(object_[1], stringType))):
+                return True
+        elif name == "geom":
+            if (isinstance(object_, NodePath)):
+                return True
+        return False
+
+    def new_set(self, name, object_=None):
+        """my new code golf par is 1"""
```

Comment:
I'm not sure what this docstring is intended to communicate to a peruser of the API reference.